# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1335579751 14400
#      Fri Apr 27 22:22:31 2012 -0400
# Node ID 7d11812c52aa0d5bfa67c68dd98f218d25eb2648
# Parent  721442e1b82cd6f7e78597a17ce8116d2a7665d4
Properly set the legend's text color property to the legend's "textcolor"
property value.  (Bug # 36136)

legend.m (updatelegendtext): Set the "color" of the text to that specified by
the "textcolor" property.

diff --git a/scripts/plot/legend.m b/scripts/plot/legend.m
--- a/scripts/plot/legend.m
+++ b/scripts/plot/legend.m
@@ -814,19 +814,17 @@ function updatelegend (h, d)
 endfunction
 
 function updatelegendtext (h, d)
   hax = get (h, "userdata").handle;
   kids = get (h, "children");
   text_kids = findobj (kids, "-property", "interpreter", "type", "text");
   interpreter = get (h, "interpreter");
   textcolor = get (h, "textcolor");
-  set (kids, "interpreter", interpreter, "color", textcolor);
-  hobj = cell2mat (get (kids, "userdata"));
-  set (hobj, "interpreter", interpreter);
+  set (text_kids, "interpreter", interpreter, "color", textcolor);
 endfunction
 
 function hideshowlegend (h, d, ca, pos1, pos2)
   isvisible = strcmp (get (h, "visible"), "off");
   if (! isvisible)
     kids = get (h, "children");
     for i = 1 : numel (kids)
       if (! strcmp (get (kids(i), "visible"), "off"))
